react/javascript/src/components/gherkin/HookStep.tsx

```diff
@@ -20,11 +20,13 @@ const HookStep: React.FunctionComponent<IProps> = ({ step }) => {
   const attachments = cucumberQuery.getTestStepsAttachments([step.id])
 
   if (stepResult.status === 'FAILED') {
-    const location = hook.sourceReference.location
-      ? hook.sourceReference.uri + ':' + hook.sourceReference.location.line
-      : hook.sourceReference.javaMethod
-      ? hook.sourceReference.javaMethod.className + '.' + hook.sourceReference.javaMethod.methodName
-      : 'Unknown location'
+    let location = 'Unknown location'
+    if (hook && hook.sourceReference.location) {
+      location = `${hook.sourceReference.uri}:${hook.sourceReference.location.line}`
+    } else if (hook && hook.sourceReference.javaMethod) {
```

Comment:
```suggestion
    if (hook && hook?.sourceReference?.location) {
      location = `${hook.sourceReference.uri}:${hook.sourceReference.location.line}`
    } else if (hook && hook?.sourceReference?.javaMethod) {
```
Just in case?

Reply:
Implemented part of this, but with `sourceReference` according to the messages schema it must always be present on a `Hook`, so should be safe to omit the null check there?

Reply:
Yes, maybe

I am always puzzled when trying to find the good balance between code safety vs being too much paranoid 😑